commit d57db9d9f5bb1617c5874242cb3a5b40cdc2a36c
Author: Wolfgang Rohdewald <wolfgang@rohdewald.de>
Date:   Mon Sep 15 15:25:41 2014 +0200

    Ticket 7665: test.test_banana: eliminate duplicate code and fix wrong assertion message
    
    This also fixes a future problem with porting to Python 3:
    StringIO.StringIO.truncate(0) implicitly does seek(0) first
    but io.bytesIO.truncate(0) does not

diff --git twisted/spread/topfiles/7665.misc twisted/spread/topfiles/7665.misc
new file mode 100644
index 0000000..a73f400
--- /dev/null
+++ twisted/spread/topfiles/7665.misc
@@ -0,0 +1 @@
+test.test_banana: eliminated duplicate code and fixed an assertion message
diff --git twisted/test/test_banana.py twisted/test/test_banana.py
index d7c5d3e..d9f5393 100644
--- twisted/test/test_banana.py
+++ twisted/test/test_banana.py
@@ -39,6 +39,15 @@ class BananaTestCase(unittest.TestCase):
         self.enc.connectionLost(failure.Failure(main.CONNECTION_DONE))
         del self.enc
 
+    def _encoded(self, n):
+        """
+        Helper: returns encoded string for n
+        """
+        self.io.seek(0)
+        self.io.truncate()
+        self.enc.sendEncoded(n)
+        return self.io.getvalue()
+
     def testString(self):
         self.enc.sendEncoded("hello")
         l = []
@@ -207,9 +216,7 @@ class BananaTestCase(unittest.TestCase):
             for add in (0, 1):
                 m = 2 ** exp + add
                 for n in (m, -m-1):
-                    self.io.truncate(0)
-                    self.enc.sendEncoded(n)
-                    self.enc.dataReceived(self.io.getvalue())
+                    self.enc.dataReceived(self._encoded(n))
                     self.assertEqual(self.result, n)
                     if n > sys.maxint or n < -sys.maxint - 1:
                         self.assertIsInstance(self.result, long)
@@ -321,7 +328,7 @@ class BananaTestCase(unittest.TestCase):
         foo = [1, 2, [3, 4], [30.5, 40.2], 5, ["six", "seven", ["eight", 9]], [10], []]
         self.enc.sendEncoded(foo)
         self.enc.dataReceived(self.io.getvalue())
-        assert self.result == foo, "%s!=%s" % (repr(self.result), repr(self.result))
+        assert self.result == foo, "%s!=%s" % (repr(self.result), repr(foo))
 
     def testPartial(self):
         foo = [1, 2, [3, 4], [30.5, 40.2], 5,
@@ -329,8 +336,7 @@ class BananaTestCase(unittest.TestCase):
                # TODO: currently the C implementation's a bit buggy...
                sys.maxint * 3l, sys.maxint * 2l, sys.maxint * -2l]
         self.enc.sendEncoded(foo)
-        for byte in self.io.getvalue():
-            self.enc.dataReceived(byte)
+        self.feed(self.io.getvalue())
         assert self.result == foo, "%s!=%s" % (repr(self.result), repr(foo))
 
     def feed(self, data):
@@ -383,34 +389,28 @@ class BananaTestCase(unittest.TestCase):
         serialized as C{INT} or C{NEG} and that larger integers are
         serialized as C{LONGINT} or C{LONGNEG}.
         """
-        def encoded(n):
-            self.io.seek(0)
-            self.io.truncate()
-            self.enc.sendEncoded(n)
-            return self.io.getvalue()
-
         baseIntIn = +2147483647
         baseNegIn = -2147483648
 
         baseIntOut = '\x7f\x7f\x7f\x07\x81'
-        self.assertEqual(encoded(baseIntIn - 2), '\x7d' + baseIntOut)
-        self.assertEqual(encoded(baseIntIn - 1), '\x7e' + baseIntOut)
-        self.assertEqual(encoded(baseIntIn - 0), '\x7f' + baseIntOut)
+        self.assertEqual(self._encoded(baseIntIn - 2), '\x7d' + baseIntOut)
+        self.assertEqual(self._encoded(baseIntIn - 1), '\x7e' + baseIntOut)
+        self.assertEqual(self._encoded(baseIntIn - 0), '\x7f' + baseIntOut)
 
         baseLongIntOut = '\x00\x00\x00\x08\x85'
-        self.assertEqual(encoded(baseIntIn + 1), '\x00' + baseLongIntOut)
-        self.assertEqual(encoded(baseIntIn + 2), '\x01' + baseLongIntOut)
-        self.assertEqual(encoded(baseIntIn + 3), '\x02' + baseLongIntOut)
+        self.assertEqual(self._encoded(baseIntIn + 1), '\x00' + baseLongIntOut)
+        self.assertEqual(self._encoded(baseIntIn + 2), '\x01' + baseLongIntOut)
+        self.assertEqual(self._encoded(baseIntIn + 3), '\x02' + baseLongIntOut)
 
         baseNegOut = '\x7f\x7f\x7f\x07\x83'
-        self.assertEqual(encoded(baseNegIn + 2), '\x7e' + baseNegOut)
-        self.assertEqual(encoded(baseNegIn + 1), '\x7f' + baseNegOut)
-        self.assertEqual(encoded(baseNegIn + 0), '\x00\x00\x00\x00\x08\x83')
+        self.assertEqual(self._encoded(baseNegIn + 2), '\x7e' + baseNegOut)
+        self.assertEqual(self._encoded(baseNegIn + 1), '\x7f' + baseNegOut)
+        self.assertEqual(self._encoded(baseNegIn + 0), '\x00\x00\x00\x00\x08\x83')
 
         baseLongNegOut = '\x00\x00\x00\x08\x86'
-        self.assertEqual(encoded(baseNegIn - 1), '\x01' + baseLongNegOut)
-        self.assertEqual(encoded(baseNegIn - 2), '\x02' + baseLongNegOut)
-        self.assertEqual(encoded(baseNegIn - 3), '\x03' + baseLongNegOut)
+        self.assertEqual(self._encoded(baseNegIn - 1), '\x01' + baseLongNegOut)
+        self.assertEqual(self._encoded(baseNegIn - 2), '\x02' + baseLongNegOut)
+        self.assertEqual(self._encoded(baseNegIn - 3), '\x03' + baseLongNegOut)
 
 
     def test_typeNotImplemented(self):
@@ -432,10 +432,7 @@ class BananaTestCase(unittest.TestCase):
         self.io.truncate()
         self.enc.dataReceived(b'\x10\x87')
         self.assertEqual(self.result, b'remote')
-        self.io.seek(0)
-        self.io.truncate()
-        self.enc.sendEncoded(b'lcache')
-        self.enc.dataReceived(self.io.getvalue())
+        self.enc.dataReceived(self._encoded(b'lcache'))
         self.assertEqual(self.result, b'lcache')
 
 
